packages/edit-post/src/components/keyboard-shortcuts/index.js

```diff
@@ -70,15 +70,18 @@ function KeyboardShortcuts() {
 		if ( blockName !== 'core/paragraph' && blockName !== 'core/heading' ) {
 			return;
 		}
-		const currentAttributes = getBlockAttributes( currentClientId );
-		const { content: currentContent, align: currentAlign } =
-			currentAttributes;
+		const attributes = getBlockAttributes( currentClientId );
+		const textAlign =
+			blockName === 'core/paragraph' ? 'align' : 'textAlign';
+		const destinationTextAlign =
+			destinationBlockName === 'core/paragraph' ? 'align' : 'textAlign';
+
 		replaceBlocks(
 			currentClientId,
 			createBlock( destinationBlockName, {
 				level,
-				content: currentContent,
-				align: currentAlign,
+				content: attributes.content,
+				...{ [ destinationTextAlign ]: attributes[ textAlign ] },
```

Comment:
Heading blocks and paragraph blocks have different property names for text alignment:

- Heading Block: `textAlign`
- Paragraph Block: `align`

Therefore, property names and values must be properly mapped.